File: src/main/java/com/example/smarthouse/sensor/DefaultSensor.java
package com.example.smarthouse.sensor;

import com.example.smarthouse.sensor.Sensor;
import com.example.smarthouse.sensor.config.SensorProtocol;
import com.example.smarthouse.sensor.config.SensorState;
import com.example.smarthouse.sensor.config.SensorType;
import com.example.smarthouse.sensor.message.MessageType;
import com.example.smarthouse.sensor.message.SensorMessage;
import lombok.Data;

import java.util.Date;

@Data
public abstract class DefaultSensor implements Sensor {
    private final int sensorId;
    private final String sensorName;
    private SensorState sensorState;
    private final SensorProtocol sensorProtocol;
    private final SensorType sensorType;

    public DefaultSensor(int sensorId, String sensorName, SensorState sensorState, SensorProtocol sensorProtocol, SensorType sensorType) {
        this.sensorId = sensorId;
        this.sensorName = sensorName;
        this.sensorState = sensorState;
        this.sensorProtocol = sensorProtocol;
        this.sensorType = sensorType;
    }

    @Override
    public SensorMessage getDataFromSensor() {
        return new SensorMessage(
                sensorId,
                0, // Placeholder for sensor data, to be implemented in subclasses
                new Date(),
                MessageType.SENSOR_DATA,
                null,
                sensorType
        );
    }

    @Override
    public void switchSensorState() {
        sensorState = (sensorState == SensorState.ON) ? SensorState.OFF : SensorState.ON;
    }
}

File: src/main/kotlin/com/example/DefaultSensor.kt
package com.example.smarthouse.sensor

import com.example.command.CommandType
import com.example.config.SensorProtocol
import com.example.config.SensorState
import com.example.config.SensorType
import com.example.message.MessageType
import com.example.message.Message
import java.util.*

abstract class DefaultSensor(
    private val sensorId: Int,
    private val sensorName: String?,
    private var sensorState: SensorState,
    private val sensorProtocol: SensorProtocol?,
    private val sensorType: SensorType
) {
    open val dataFromSensor: Message
        get() = Message(
            sensorId,
            0,
            Date(),
            MessageType.SENSOR_DATA,
            null,
            sensorType
        )

    fun getSensorId(): Int {
        return sensorId
    }

    fun getSensorName(): String? {
        return sensorName
    }

    fun getSensorState(): SensorState {
        return sensorState
    }

    fun setSensorState(sensorState: SensorState) {
        this.sensorState = sensorState
    }

    fun getSensorProtocol(): SensorProtocol? {
        return sensorProtocol
    }

    fun getSensorType(): SensorType {
        return sensorType
    }

    open fun executeCommand(command: CommandType, parameters: Any? = null): Message? {
        // Здесь можно реализовать выполнение команды
        return null
    }
}



